File: native/android/PropertyCross/src/com/propertycross/android/presenter/searchitems/GeoLocationSearchItem.java
package com.propertycross.android.presenter.searchitems;

import com.propertycross.android.events.Callback;
import com.propertycross.android.model.PropertyDataSource;
import com.propertycross.android.model.PropertyDataSourceResult;
import com.propertycross.android.presenter.GeoLocation;

public class GeoLocationSearchItem extends SearchItem {
	private GeoLocation location;
	private String displayText;

	public GeoLocationSearchItem() {
	}

	public GeoLocationSearchItem(GeoLocation location) {
		this.location = location;
		this.displayText = String.format("{0:F2}, {1:F2}",
				new Object[] {
					Double.valueOf(location.getLatitude()),
					Double.valueOf(location.getLongitude())
				});
	}

	public String getDisplayText() {
		return displayText;
	}

	public void setDisplayText(String text) {
		displayText = text;
	}

	public GeoLocation getLocation() {
		return location;
	}

	public void setLocation(GeoLocation location) {
		this.location = location;
	}

	public void findProperties(PropertyDataSource source, int pageNumber,
			Callback<PropertyDataSourceResult> complete, Callback<Exception> error) {
		source.findProperties(location.getLatitude(), location.getLongitude(), pageNumber, complete, error);
	}
}

File: xamarin-Mvvmcross/PropertyCross.Core/Domain/SearchItem/GeoLocationSearchItem.cs
﻿using System;
using PropertyCross.Core.Domain.Model;
using PropertyCross.Core.Domain.Services;

namespace PropertyCross.Core.Domain.SearchItem
{
  /// <summary>
  /// Represents a search against a geo-location.
  /// </summary>
  public class GeoLocationSearchItem : SearchItemBase
  {
    public GeoLocation Location { get; set; }

    public GeoLocationSearchItem()
    {
    }

    public GeoLocationSearchItem(GeoLocation location)
    {
      Location = location;
      DisplayText = string.Format("{0:F2}, {1:F2}", location.Latitude, location.Longitude);
    }

    public override void FindProperties(PropertyDataSource dataSource,
      int pageNumber, Action<PropertyDataSourceResult> callback, Action<Exception> error)
    {
      dataSource.FindProperties(Location.Latitude, Location.Longitude, pageNumber, callback, error);
    }
  }
}
